### ISSUE TITLE
API for marking Grid cells as row headers

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/platform#3738]
## Description
New API in Grid for defining which columns' cells should be marked up as headers for their rows, in order to improve usability with screen readers. See https://github.com/vaadin/web-components/issues/5321 for more details.

## Use cases
As a screen reader user
I want Grid cell contents to be announced in the context of the row that they're in
So that it's easier for me to understand the contents of a Grid

## Acceptance criteria
- [ ] Java API for setting a column as row header: `Column.setRowHeader(boolean isRowHeader)`
- [ ] WC API for setting a column as row header: `<vaadin-grid-column row-header>`
- [ ] Table body cells in columns marked as row headers are rendered as `<td role="rowheader" ...>` instead of `<td role="gridcell" ...>`
- [ ] All other attributes applied to normal cells must be retained
- [ ] Multiple columns can be set as row-headers
- [ ] Any columns can be set as row-headers
- [ ] This change should only affect body cells, not header or footer cells.
- [ ] Supporting run-time modification of row-header settings is optional (it's acceptable that the columns need to be removed and re-added to accomplish this)

## General criteria
- [x] APIs reviewed
- [ ] Performance 
- [ ] UX/DX tests in Alpha 
- [ ] Documentation: 
- [ ] How to test? 
- [ ] Limitations: 

## Security
- [x] Security implications have been taken into account (elaborate or link to product security requirement specification if there are implications)

**Comment by czp13** [comment]
Sorry, I was full of developments, and work in the past 3-4 weeks, and also off for 1 week and some days. 
I try to pick this up ASAP. This is not forgotten.

**Comment by web-padawan** [comment]
@czp13 No worries, there is plenty of time before Vaadin 24.1 and we didn't even release the first alpha yet 🙂

**Comment by czp13** [comment]
@web-padawan: thank you, that is comforting to know :)

**Comment by czp13** [comment]
I finish my flow 6  weeks + cooldown weeks soonish so I will have more **time** finally after the builder challenge or even a bit before I plan to pick this up finally! 🙇
